parser.js

```diff
@@ -73,17 +77,31 @@ function generateAST(code, options, additionalParsingContext) {
         if (typeof options.tokens === "boolean" && options.tokens) {
             extra.tokens = [];
         }
+
         if (typeof options.comment === "boolean" && options.comment) {
             extra.comment = true;
             extra.comments = [];
         }
+
         if (typeof options.tolerant === "boolean" && options.tolerant) {
             extra.errors = [];
         }
 
-        if (options.ecmaFeatures && typeof options.ecmaFeatures === "object") {
-            // pass through jsx option
-            extra.ecmaFeatures.jsx = options.ecmaFeatures.jsx;
+        const hasEcmaFeatures = options.ecmaFeatures && typeof options.ecmaFeatures === "object";
+        const hasFilePath = options.filePath !== DEFAULT_FILEPATH;
```

Comment:
Actually, should this be `options.filePath && (options.filePath !== DEFAULT_FILEPATH)`? If no file path is provided, this would currently still resolve to true.